dataline-workers/src/main/java/io/dataline/workers/singer/SingerTarget.java

```diff
@@ -27,56 +27,72 @@
 import com.fasterxml.jackson.databind.JsonNode;
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Charsets;
-import io.dataline.commons.functional.CloseableConsumer;
+import com.google.common.base.Preconditions;
 import io.dataline.commons.io.IOs;
 import io.dataline.commons.json.Jsons;
 import io.dataline.config.StandardTargetConfig;
 import io.dataline.singer.SingerMessage;
-import io.dataline.workers.DefaultSyncWorker;
-import io.dataline.workers.TargetConsumer;
-import io.dataline.workers.TargetFactory;
+import io.dataline.workers.WorkerUtils;
 import io.dataline.workers.process.ProcessBuilderFactory;
 import java.io.BufferedWriter;
+import java.io.IOException;
 import java.io.OutputStreamWriter;
 import java.nio.file.Path;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class SingerTargetFactory implements TargetFactory<SingerMessage> {
+public class SingerTarget {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(SingerTargetFactory.class);
+  private static final Logger LOGGER = LoggerFactory.getLogger(SingerTarget.class);
 
   @VisibleForTesting
   static final String CONFIG_JSON_FILENAME = "target_config.json";
 
   private final String imageName;
   private final ProcessBuilderFactory pbf;
 
-  public SingerTargetFactory(final String imageName, final ProcessBuilderFactory pbf) {
+  private Process targetProcess;
+  private BufferedWriter writer;
+
+  public SingerTarget(final String imageName, final ProcessBuilderFactory pbf) {
     this.imageName = imageName;
     this.pbf = pbf;
   }
 
-  @Override
-  public CloseableConsumer<SingerMessage> create(StandardTargetConfig targetConfig, Path jobRoot) {
+  public void start(StandardTargetConfig targetConfig, Path jobRoot) {
+    Preconditions.checkState(targetProcess == null);
+
     final JsonNode configDotJson = targetConfig.getDestinationConnectionImplementation().getConfiguration();
 
     // write config.json to disk
     IOs.writeFile(jobRoot, CONFIG_JSON_FILENAME, Jsons.serialize(configDotJson));
 
     try {
       LOGGER.info("Running Singer target...");
-      final Process targetProcess =
+      targetProcess =
           pbf.create(jobRoot, imageName, "--config", CONFIG_JSON_FILENAME)
-              .redirectError(jobRoot.resolve(DefaultSyncWorker.TARGET_ERR_LOG).toFile())
+              .redirectError(jobRoot.resolve(SingerSyncWorker.TARGET_ERR_LOG).toFile())
               .start();
 
-      // the TargetConsumer is responsible for closing this.
-      BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(targetProcess.getOutputStream(), Charsets.UTF_8));
-      return new TargetConsumer(writer, targetProcess);
+      writer = new BufferedWriter(new OutputStreamWriter(targetProcess.getOutputStream(), Charsets.UTF_8));
     } catch (Exception e) {
       throw new RuntimeException(e);
     }
   }
 
+  public void consume(SingerMessage message) throws IOException {
```

Comment:
This is the code from the TargetConsumer

Reply:
same. if this is a consumer, why aren't we using that interface?